flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/split/PulsarOrderedPartitionSplitReader.java

```diff
@@ -75,7 +78,30 @@ protected void finishedPollMessage(Message<byte[]> message) {
 
     @Override
     protected void startConsumer(PulsarPartitionSplit split, Consumer<byte[]> consumer) {
-        initialStartPosition(split, consumer);
+        MessageId latestConsumedId = split.getLatestConsumedId();
+
+        // Reset the start position for ordered pulsar consumer.
+        if (latestConsumedId != null) {
+            StartCursor startCursor = StartCursor.fromMessageId(latestConsumedId, false);
+            TopicPartition partition = split.getPartition();
+
+            try {
+                startCursor.seekPosition(
+                        partition.getTopic(), partition.getPartitionId(), consumer);
```

Comment:
Why do we seek here? I thought the purpose was to only seek in enumerator?

Reply:
It's used only for `Exclusive` or `Failover` subscriptions. Only one consumer would be used for seeking the start position.

Reply:
In case of restart, isn't `latestConsumedId` always `null`? I thought we would update the `startCursor` on checkpoint, such that it should be properly initialized in enumerator/reader right?

Reply:
I have changed the serialization logic, the `latestConsumedId` shouldn't be `null` here when we restart.